src/mono/browser/runtime/managed-exports.ts

```diff
@@ -374,21 +374,16 @@ function bind_assembly_exports_mono (assemblyName: string): Promise<void> {
     }
 }
 
-export const exportsByAssembly: Map<string, any> = new Map();
 function bind_assembly_exports_naot (assembly: string) {
-    assert_js_interop();
-    const result = exportsByAssembly.get(assembly);
-    if (!result) {
-        let assemblyWithoutExtension = assembly;
-        if (assemblyWithoutExtension.endsWith(".dll")) {
-            assemblyWithoutExtension = assemblyWithoutExtension.substring(0, assembly.length - 4);
-        }
-        const register = (Module as any)["_" + assemblyWithoutExtension + "__GeneratedInitializer" + "__Register_"];
-        mono_assert(register, `Missing wasm export for JSExport registration function in assembly ${assembly}`);
-        register();
+    loaderHelpers.assert_runtime_running();
+    let assemblyWithoutExtension = assembly;
+    if (assemblyWithoutExtension.endsWith(".dll")) {
```

Comment:
no webcil ?

Reply:
What are mono users passing as assembly extension for JSExports?

Reply:
I have not realized this is coming from the user. Perhaps we need similar logic for mono too ?

Reply:
Here we need to strip the extension for get a name for wasm export. Whether extension should be optional or not on mono is a different question. But based on lazy loading, we probably should be supporting extension agnostic assembly names